### DISCUSSION TITLE
in CommonJS，UMD， compile ，Splitting the code will result in an error。

### DISCUSSION THREAD

**Discussion by ygs-code** [webpack/webpack#16132] [General]
# Bug report

**What is the current behavior?**

**If the current behavior is a bug, please provide the steps to reproduce.**

**What is the expected behavior?**

**Other relevant information:**
webpack version:    4 or  5
Node.js version:  16
Operating System:  mac or windows
Additional tools:

I was packaging the Node service code，I use，  libraryTarget: "umd", or   libraryTarget: "commonJS",  
 If you are using splitChunks  or  webpack.optimize.CommonsChunkPlugin  or  Dynamic Imports  () => import('@/views/login') ，When running JS, an error is reported when referring to the split code。 This is a problem in both  has webpack4 and 5 。

**Comment by vankop** [comment]
please create small reproducible repo

**Comment by ygs-code** [comment]
I used the setTimeout timer to do an asynchronous import of C files, but I found that the import of C files would get an error.

Error: Cannot find module './static/js/1.js'
Require stack:

code ：
a.js 
```
import b from './b'
b()
```

b.js
```
export default () => {
    console.log('bbbbbbbbbbbbbbbb');
    setTimeout(async () => {
        const c = import('./c');
        await c();
    }, 2000);
};

```

c.js
```
export default () => {
    console.log('ccccccccccccc');
};

```

webpack   config ；
```
const path = require('path');
module.exports = {
    name: 'server',
    target: 'node',
    node: {
        __filename: true,
        __dirname: true,
        global: false,
    },
    entry: {
        a: path.join(__dirname, '/a.js'),
    },
    output: {
        filename: 'static/js/[name].js',
        path: path.join(__dirname, '/dist'),
        publicPath: '/',
        chunkFilename: 'static/js/[name].js',
     
        // strictModuleExceptionHandling: !isEnvProduction,
        // 导出库(exported library)的名称
        library: 'server',
        //   导出库(exported library)的类型
        libraryTarget: 'umd',
        // libraryTarget: 'commonjs2',
        // 在 UMD 库中使用命名的 AMD 模块
        umdNamedDefine: true,
        globalObject: 'this',
    },
    //在第一个错误出现时抛出失败结果，而不是容忍它
    bail: true,
    module: {
        rules: [
            {
                test: /(\.jsx?$)|(\.js?$)/,
                exclude: /node_modules/,
                // include: path.resolve(rootPath, 'client'),
                use: {
                    loader: 'babel-loader',
                    options: {
                        // presets: ['env', 'react', 'stage-0'],
                        // plugins: ['transform-runtime', 'add-module-exports'],
                        cacheDirectory: true,
                    },
                },
            },
        ],
    },
};

```


After running Webpack compile . 

run
```
  node   ./dist/static/js/a.js
```
After 1 second
Error: Cannot find module './static/js/1.js'
Require stack:

**Comment by ygs-code** [comment]
It depends on how I configure it. If I configure it this way, it's fine  
```
{
  filename: '[name].js',  
  chunkFilename: '[name].js',
}  
```

But I want to add the static/js directory in front of the JS file so that I can better distinguish between JS, HTML and CSS 
files. 。 

Then I also have a problem configuring Webpack this way
```
  {
  filename: 'static/js/[name].js',   
  chunkFilename: '[name].js',
} 
```

**Comment by vankop** [comment]
problem in `publicPath`, try to use `publicPath: auto`
